pyccel/codegen/printing/cucode.py

```diff
@@ -131,4 +137,44 @@ def _print_ModuleHeader(self, expr):
                           global_variables,
                           function_declaration,
                           "#endif // {name.upper()}_H\n"))
+    def _print_Allocate(self, expr):
+        variable = expr.variable
+        if not isinstance(variable.class_type, CudaArrayType):
+            return super()._print_Allocate(expr)
+        shape = ", ".join(self._print(i) for i in expr.shape)
+        if isinstance(variable.class_type, CudaArrayType):
+            dtype = self.find_in_ndarray_type_registry(variable.dtype)
+        else:
+            raise NotImplementedError(f"Don't know how to index {variable.class_type} type")
+        shape_Assign = f"int64_t shape_Assign_{expr.variable.name} [] = {{{shape}}};\n"
+
+        is_view = 'false' if variable.on_heap else 'true'
+        memory_location = variable.class_type.memory_location
+        if memory_location in ('device', 'host'):
+            memory_location = str(memory_location).capitalize() + 'Memory'
+        else:
+            memory_location = 'managedMemory'
+        self.add_import(cu_imports['cuda_ndarrays'])
+        alloc_code = f"{self._print(expr.variable)} = cuda_array_create({variable.rank},  shape_Assign_{expr.variable.name}, {dtype}, {is_view},{memory_location});\n"
+        return f'{shape_Assign} {alloc_code}'
+
+    def _print_Deallocate(self, expr):
+        var_code = self._print(expr.variable)
+
+        if not isinstance(expr.variable.class_type, CudaArrayType):
+            return super()._print_Deallocate(expr)
+
+        if expr.variable.class_type.memory_location == 'host':
+            return f"cuda_free_host({var_code});\n"
+        else:
+            return f"cuda_free({var_code});\n"
+
+    def _print_Assign(self, expr):
+        rhs = expr.rhs
+        if isinstance(rhs.class_type, CudaArrayType):
+            if(isinstance(rhs, (CudaFull))):
+            # TODO add support for CudaFull
+                return " \n"
```

Comment:
Is it not safer to not include this code so that the neat error is raised instead of just printing nothing?